src/audio/pipeline/pipeline-stream.c

```diff
@@ -367,6 +375,9 @@ int pipeline_trigger(struct pipeline *p, struct comp_dev *host, int cmd)
 {
 	int ret;
 #if CONFIG_KCPS_DYNAMIC_CLOCK_CONTROL
+/* FIXME: this must be a platform-specific parameter or a Kconfig option */
+#define DSP_MIN_KCPS 50000
```

Comment:
There is no point in keeping KCPS feature with such limit because it means we cannot go lower than highest available clock which is not true because there are basic topologies which we can handle on 38.4.

So why is this change required for llext module?

Reply:
@abonislawski what do you mean "lower than highest available clock?" Our highest available clock is hundreds of MHz, i.e. hundreds of KCPS, isn't it? So this is definitely lower than that.
This patch is needed not only for LLEXT, I think there's a bug in general in the current `CONFIG_KCPS_DYNAMIC_CLOCK_CONTROL` implementation - see commit description. Using a "safe" value for modules with a 0 CPC isn't implemented correctly. After I've fixed it I started getting lock ups when destroying pipelines. E.g. I had cases when the system was starting with 20000 KCPS, then a pipeline was constructed with a 0-CPC module, so the KCPS went to the maximum. Then as the pipeline was destroyed I tried to return to 20000 KCPS and then I got IPC timeouts in the driver. Introducing a minimum of 50000 (or 40000 worked too) KCPS fixed the problem.

Reply:
KCPS higher than 38400 will switch to our highest clock and _DSP_MIN_KCPS_ is already higher than that. _PRIMARY_CORE_BASE_CPS_USAGE_ is currently 20000 (higher than needed).
So the one thing is to fix 0CPC problem (which I thought was fixed with previous versions of this commit) but another thing is to introduce such limit like _DSP_MIN_KCPS_ which I dont understand